src/CustomHooks/RandomArticles.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect } from "react";
+import { getArticles } from "../Api/Api";
+
+const RandomArticles = ({ articles, randomFiveArticles }) => {
+  const [articleList, setArticleList] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  const fetchArticles = async () => {
```

Comment:
Do you need to fetch all articles again here? You are passing the articles on props - otherwise 2 fetches are occuring